sklearn/ensemble/stacking.py

```diff
@@ -0,0 +1,179 @@
+
+    def fit_transform(self, X, y, **fit_params):
+        preds = cross_val_predict(self.base_estimator, X, y, cv=self.cv,
+                                  method=self._method_name(),
+                                  n_jobs=self.n_jobs, fit_params=fit_params)
+
+        self.base_estimator.fit(X, y, **fit_params)
+
+        if preds.ndim == 1:
+            preds = preds.reshape(-1, 1)
+
+        return preds
+
+
+def _identity_transformer():
+    """Contructs a transformer that does nothing"""
+    return FunctionTransformer(lambda x: x)
+
+
+def make_stack_layer(*base_estimators, **kwargs):
+    """Construct a single layer for a stacked model.
+
+    This is a wrapper around pipelines to provide a more convenient API for
+    stacking models.
+
+    Parameters
+    ----------
+    *base_estimators : list of base estimators.
+
+    restacking : bool, optional (default=False)
+        When true, the transformer will return the input.
+
+    transformer_weights : dict, optional (default=None)
+        Multiplicative weights for features per transformer.
+        Keys are transformer names, values the weights.
+
+    **kwargs : Keyword arguments to be passed to `StackMetaEstimator`.
+
+    Returns
+    -------
+    f : FeatureUnion with every base estimator wrapped in a
+         `StackMetaEstimator`.
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from sklearn.ensemble import make_stack_layer
+    >>> from sklearn.neighbors import KNeighborsClassifier
+    >>> from sklearn.svm import SVC
+    >>> t = make_stack_layer(KNeighborsClassifier(n_neighbors=3), SVC())
+    >>> X = np.array([[1, 3], [.12, 1], [.5, -2], [1, -1], [-2, .1], [7, -84]])
+    >>> y = np.array([1, 0, 0, 1, 0, 1])
+    >>> t.fit_transform(X, y) # doctest: +NORMALIZE_WHITESPACE
+    array([[ 0.66666667,  0.33333333,  0.        ],
+           [ 0.66666667,  0.33333333,  1.        ],
+           [ 0.66666667,  0.33333333,  1.        ],
+           [ 0.66666667,  0.33333333,  0.        ],
+           [ 0.66666667,  0.33333333,  0.        ],
+           [ 0.66666667,  0.33333333,  0.        ]])
+    """
+    restacking = kwargs.pop('restacking', False)
+    transformer_weights = kwargs.pop('transformer_weights', None)
+
+    named_estimators = [(n, StackMetaEstimator(estimator, **kwargs))
+                        for n, estimator in _name_estimators(base_estimators)]
+    if restacking:
+        named_estimators.extend(_name_estimators([_identity_transformer()]))
+
+    return FeatureUnion(named_estimators,
+                        transformer_weights=transformer_weights)
+
+
+def stack_estimators(estimators, meta_estimator, **kwargs):
+    """Construct a stacked estimator
+
+    This is a wrapper around pipelines to provide a more convenient API for
+    stacking models. Estimators in `estimators_matrix` are wrapped in
+    `StackMetaEstimator`.
+
+    Parameters
+    ----------
+    estimators : 2D array with base estimators. Each row will be turned into a
```

Comment:
2d array would require the same size for each layer. I think you just want a list of lists. But I'd appreciate if a single list for a single layer worked too.